src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs

```diff
@@ -399,6 +399,38 @@ public override void Abort(ConnectionAbortedException abortReason)
             Output.CancelPendingRead();
         }
 
+        public void AbortRead(long errorCode, ConnectionAbortedException abortReason)
+        {
+            lock (_shutdownLock)
+            {
+                if (_stream.CanRead)
+                {
+                    _log.StreamAbortRead(this, abortReason.Message);
```

Comment:
If there's an active reader, can we throw this ConnectionAbortedException from the current read similar to what we do with SocketConnection?

Reply:
I'll add a test and see what happens.